DataDog/datadog-operator
# Standardize image defaulting

### What does this PR do?

Update defaulting images logic and have a dedicated package for defaulting.

The idea here is to update the Agent and Cluster-Agent image version with 
the lastest stable version each time we release the Operator. 

### Motivation

Keep up-to-date the agent and cluster-agent when deploying a new Operator version.

### Additional Notes

N/A

### Describe your test plan

Deploy the example `examples/datadogagent/datadog-agent-with-credential-secret.yaml` 

* Check that the Agent image tag is `7.30.0` 
* Check that the Cluster-Agent image tag is `1.14.0`
